diff --git a/src/Table.jsx b/src/Table.jsx
index 88ef718..779c2c8 100644
--- a/src/components/Table.jsx
+++ b/src/components/Table.jsx
@@ -28,17 +28,23 @@ export default class Table extends React.Component {

   buildRows() {
     return this.props.rows.map((row) => {
-      return (
-        <tr key={row.id}>
-          {
-            this.props.columns.map((col) => {
-              return (
-                <td key={col.key}>{row[col.key]}</td>
-              );
-            })
-          }
-        </tr>
-      );
+      return <Row key={row.id} data={row} columns={this.props.columns} />;
     });
   }
 }
+
+class Row extends React.PureComponent {
+  render() {
+    return (
+      <tr>
+        {
+          this.props.columns.map((col) => {
+            return (
+              <td key={col.key}>{this.props.data[col.key]}</td>
+            );
+          })
+        }
+      </tr>
+    );
+  }
+}
