diff --git a/src/components/Controls.jsx b/src/components/Controls.jsx
index a7c6b32..7aa6b9e 100644
--- a/src/components/Controls.jsx
+++ b/src/components/Controls.jsx
@@ -2,7 +2,7 @@ import React from 'react';
 
 import styles from './Controls.module.css';
 
-export default function Controls(props) {
+export default React.memo(function Controls(props) {
   return (
     <div className={styles.root}>
       <label>
@@ -14,4 +14,4 @@ export default function Controls(props) {
       </label>
     </div>
   );
-}
+});
